### ISSUE TITLE
Inaccessible show/hide functionality in TOC and sidebar

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3446]
The show/hide functionality that is used for the sidebar in mobile view and the table of contents (TOC)
* [ ] cannot be operated through keyboard alone and
* [ ] cannot be programmatically determined.

That means screen reader users won't understand that and how these components toggle the visibility of the content below. And people with motor impairments who cannot use the mouse or who use voice control will not be able to use the toggle. 

That fails [2.1.1 Keyboard](https://www.w3.org/TR/WCAG21/#keyboard) and [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value).

This has a high priority in the case of the sidebar as it will often have navigational links without which you might not be able to navigate the website when it's always closed for some people. (Every page is still available through the sitemap/index. But that might not be obvious and/or easy to use.)

![](https://user-images.githubusercontent.com/108893/111969206-e3e32e00-8af1-11eb-9129-1ec2fd609f41.png)

### Keyboard

The keyboard issue should ideally be fixed by using the appropriate semantic HTML element, in this case that would be a button. It doesn't need to look like a button. The button can either be around the whole heading content or could be a separate "show/hide" button.

There are other solutions, like adding a `tabindex="-1"`. But that gets more involved when you try to add other behaviours that buttons naturally have.

### Not programmatically determinable

The toggle not being programmatically determinable can either be fixed by making this functionality clear through code or through text.

The triangle is only visual. The text already includes "-" when the component is open and "+" when it is closed. But a) that text is hidden from screen readers (via `display: none`) and b) it is not a good way to describe in text what happens ("show" and "hide" would be better).

Via code a toggle is usually done via the [`aria-expanded`](https://www.w3.org/TR/wai-aria-1.1/#aria-expanded) attribute. There is one on the div that gets revealed, but it is implemented incorrectly. That's why the HTML Validator says:
> Element div is missing one or more of the following attributes: role.

NVDA recognises that it is clickable, VoiceOver doesn't. Neither NVDA nor VoiceOver recognise what would happen before or after you click on the heading, neither announce anything after you click it.

It is recommended to add the `aria-expanded` to the toggle button instead (which will already have the correct implicit role).

Another option is to use the HTML5 [details/summary](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) elements which have all the necessary accessibility already baked in.

## Other non-fail accessibility issues

These don't strictly fail WCAG but are still accessibility issues. Fixing them is recommended but optional for passing WCAG.

* [ ] It would help screen reader users navigate to the sidebar in desktop mode if its heading wasn't hidden from them. That could be fixed by hiding it visually via the `a11y` class instead of via `display: none`. (If a button gets added, that should be hidden from screen readers again when it is not visually showing.)

* [ ] When resizing text in Firefox with "zoom text only", the triangle toggle indicator doesn't get bigger. As that triangle is the only thing that indicates that the TOC or sidebar can be toggled, it should ideally still be very visible when increasing the text size.
